### PR TITLE
Re-test enzyme

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

**Comment by Vaibhavdixit02** [comment]
https://github.com/SciML/OptimizationBase.jl/actions/runs/9276344283/job/25523222868?pr=53#step:6:1044

**Comment by wsmoses** [comment]
@Vaibhavdixit02  can you open an issue with a MWE of the error. I'll try to get it quickly thereafter.

**Comment by wsmoses** [comment]
@Vaibhavdixit02 @ChrisRackauckas please do not drop the Enzyme extension in favor of DI here.

This will lead to both more codes erring and suboptimal performance as DI will create both unnecessary closures among other issues. I would prefer for user code not to be made to break because of an unnecessary interface intermediate

See some comments about this [tpapp/LogDensityProblemsAD.jl#29 (review)](https://github.com/tpapp/LogDensityProblemsAD.jl/pull/29#pullrequestreview-2051253729) / [tpapp/LogDensityProblemsAD.jl#29 (comment)](https://github.com/tpapp/LogDensityProblemsAD.jl/pull/29#discussion_r1597685998)

Unfortunately when I commented "Closures were one of the reasons imo that AD.jl didn't take off, so long term I think DI.jl would be well served by both not introducing them itself (my understanding is that it does this successfully), but also not forcing users to create the same closures user-side (which would create similar issues)." but the response was "DI is unlikely to support either multiple arguments or activity specification anytime soon"

x/ref https://github.com/gdalle/DifferentiationInterface.jl/issues/307

**Comment by wsmoses** [comment]
@Vaibhavdixit02 @ChrisRackauckas do you have a MWE (I didn't see one by chance). I'll try to get it fixed quickly

**Comment by Vaibhavdixit02** [comment]
I literally mentioned this to Chris like 2 hours ago haha, we are on the same page. Would you be able to help review the sparse support here? I started the effort in the previous home for this code https://github.com/SciML/Optimization.jl/pull/688/files I'll revive that

**Comment by Vaibhavdixit02** [comment]
Ref MWE let's see if the latest release still throws the same error first

**Comment by ChrisRackauckas** [comment]
This is now blocking https://github.com/SciML/SciMLSensitivity.jl/pull/1067

**Comment by wsmoses** [comment]
again, happy to help but need a MWE.

In interim just going to be plugging away at mixed activity stuff.

File: Project.toml [hunk 1]
```diff
@@ -38,7 +38,7 @@ OptimizationZygoteExt = "Zygote"
 ADTypes = "1"
 ArrayInterface = "7.6"
 DocStringExtensions = "0.9"
-Enzyme = "0.11.11, =0.12.6"
+Enzyme = "0.12"
```
Comment (Vaibhavdixit02) [inline Project.toml:41]:
```suggestion
Enzyme = "0.12.12"
```

**Comment by wsmoses** [comment]
bumping this

**Comment by Vaibhavdixit02** [comment]
I have been trying to recreate but ended up with a duplicated returns not handled error instead of the one on CI here - when running `]test` this one does show up so it's kind of tricky to figure out a MWE. The issue happens in out-of-place second hessian of a closure created here https://github.com/SciML/OptimizationBase.jl/blob/581f4f82675f3f0c100dfd15131881c59cbe53bb/ext/OptimizationEnzymeExt.jl#L368-L377 I suspect the `f2` is incorrect but have been unable to pin down the exact issue yet (I am still trying)

File: test/adtests.jl [hunk 2]
```diff
@@ -521,7 +521,7 @@ optprob.hess(H2, x0)
     @test optprob.hess(x0) == H1
     @test optprob.cons(x0) == [0.0, 0.0]
     @test optprob.cons_j([5.0, 3.0])≈[10.0 6.0; -0.149013 -0.958924] rtol=1e-6
-    @test_broken optprob.cons_h(x0) # == [[2.0 0.0; 0.0 2.0], [-0.0 1.0; 1.0 0.0]]
+    @test optprob.cons_h(x0) == [[2.0 0.0; 0.0 2.0], [-0.0 1.0; 1.0 0.0]]
```
Comment (Vaibhavdixit02) [inline test/adtests.jl:524]:
```suggestion
    @test_broken optprob.cons_h(x0) == [[2.0 0.0; 0.0 2.0], [-0.0 1.0; 1.0 0.0]]
```

**Pull request closed without merging**